modules/social_features/social_follow_user/social_follow_user.module

```diff
@@ -187,3 +212,80 @@ function social_follow_user_preprocess_flag(array &$variables): void {
     }
   }
 }
+
+/**
+ * Implements hook_form_FORM_ID_alter() for user_form().
+ */
+function social_follow_user_form_user_form_alter(array &$form, FormStateInterface $form_state): void {
+  $profile = _social_profile_get_profile_from_route();
+  $active = \Drupal::config('social_follow_user.settings')->get('status');
+
+  if ($profile instanceof ProfileInterface && $active) {
+    $following_enabled = \Drupal::service('social_follow_user.helper')
+      ->getFollowingStatus((int) $profile->get('uid')->target_id);
+
+    $form['profile_privacy']['social_profile_following_enabled'] = [
+      '#type' => 'checkbox',
+      '#title' => t('Allow members to follow me'),
+      '#default_value' => $following_enabled,
+      '#description' => t('This section allows you to turn off the ability for 
+members to follow your activity and receive notification about new content you 
+create.'),
+      '#attributes' => [
+        'data-switch' => TRUE,
+      ],
+    ];
+
+    // Add Submit function only when the data is actually editable.
+    $form['actions']['submit']['#submit'][] = '_social_follow_user_form_submit';
+  }
+}
+
+/**
+ * Form submit for user_form.
+ *
+ * @param array $form
+ *   The user edit form.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   Form state interface.
+ */
+function _social_follow_user_form_submit(array $form, FormStateInterface $form_state): void {
+  $profile = _social_profile_get_profile_from_route();
+  $profile_privacy = $form_state->getValue('profile_privacy');
+
+  if ($profile instanceof Profile) {
+    $uid = $profile->getOwnerId();
+
+    if (isset($profile_privacy['social_profile_following_enabled'])) {
+      // Save following status in the user data.
+      \Drupal::service('social_follow_user.helper')
+        ->setFollowingStatus((int) $uid, (bool) $profile_privacy['social_profile_following_enabled']);
+    }
+    // Invalidate profile cache tags.
+    Cache::invalidateTags(['profile:' . $uid]);
+  }
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_insert().
+ *
+ * @throws \Drupal\Core\Entity\EntityStorageException
+ */
+function social_follow_user_user_insert(UserInterface $user): void {
```

Comment:
@volodymyr-sydor It would be good to avoid this update hook, since ability to follow should be enabled by default for all users.